src/test/kotlin/io/github/nomisrev/service/ArticleServiceSpec.kt

```diff
@@ -0,0 +1,116 @@
+package io.github.nomisrev.service
+
+import arrow.core.flatMap
+import io.github.nefilim.kjwt.JWSHMAC512Algorithm
+import io.github.nefilim.kjwt.JWT
+import io.github.nomisrev.*
+import io.github.nomisrev.repo.UserId
+import io.kotest.assertions.arrow.core.shouldBeRight
+import io.kotest.assertions.arrow.core.shouldBeSome
+import io.kotest.matchers.ints.shouldBeExactly
+
+class ArticleServiceSpec :
+  SuspendFun({
+    val articleService: ArticleService = KotestProject.dependencies.get().articleService
+    val userService: UserService = KotestProject.dependencies.get().userService
+
+    // User
+    val kaavehUsername = "kaaveh"
+    val kaavehEmail = "kaaveh@domain.com"
+    val kaavehPw = "123456789"
+    val simonUsername = "simon"
+    val simonEmail = "simon@domain.com"
+    val simonPw = "123456789"
+    val johnUsername = "john"
+    val johnEmail = "john@domain.com"
+    val johnPw = "123456789"
+
+    // Article
+    val validTags = setOf("arrow", "ktor", "kotlin", "sqldelight")
+    val validTitle = "Fake Article Arrow "
+    val validDescription = "This is a fake article description."
+    val validBody = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
+
+    "getUserFeed" -
+      {
+        "get empty kaaveh's feed when he followed nobody" {
+          // Create users
+          val kaavehId =
+            userService
+              .register(RegisterUser(kaavehUsername, kaavehEmail, kaavehPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
+          val simonId =
+            userService
+              .register(RegisterUser(simonUsername, simonEmail, simonPw))
+              .flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
+              .map { it.claimValueAsLong("id").shouldBeSome() }
+              .shouldBeRight()
```

Comment:
I think we should split this into a top-level extension function.

```kotlin
fun Either<DomainError, JwtToken>.shouldHaveUserId(): UserId =
  flatMap { JWT.decodeT(it.value, JWSHMAC512Algorithm) }
  .map { it.claimValueAsLong("id").shouldBeSome() }
  .shouldBeRight()
```

Such that we can clean up this code as:

```kotlin
userService
              .register(RegisterUser(kaavehUsername, kaavehEmail, kaavehPw))
              . shouldHaveUserId()
```